hw/ip/otbn/doc/_index.md

```diff
@@ -426,6 +450,42 @@ These accesses are ignored if OTBN is busy.
 A host processor can check whether OTBN is busy by reading the {{< regref "STATUS.busy">}} flag.
 All memory accesses through the register interface must be word-aligned 32b word accesses.
 
+### Data Integrity Protection {#design-details-data-integrity-protection}
+
+#### Integrity Protection Code {#design-details-integrity-protection-code}
+
+OTBN uses the same integrity protection code everywhere to provide overarching data protection without regular re-encoding.
+The code is applied to 32b data words, and produces 39b of encoded data.
+
+The code used is an inverted (39,32) Hsiao "single error correction, double error detection" (SECDED) error correction code (ECC) [[CHEN08]({{< relref "#ref-chen08">}})].
+It has a minimum Hamming distance of four, resulting in the ability to detect at least three errors in a 32 bit word.
+The code is used for error detection only; no error correction is performed.
+In contrast to the original Hsiao code the outputs are inverted to produce a code in which the all-zero word (0x0) is not a valid code word.
+
+#### Register File Integrity Protection
+
+OTBN contains two register files: the 32b GPRs and the 256b WDRs.
+The data stored in both register files is protected with the [Integrity Protection Code]({{< relref "#design-details-integrity-protection-code">}}).
+
+The GPRs `x2` to `x31` store a 32b data word together with the integrity protection code, resulting in 39b of stored data.
+(`x0`, the zero register, and `x1`, the call stack, require special treatment.)
+
+Each 256b Wide Data Register (WDR) stores a 256b data word together with the integrity protection code, resulting in 312b of stored data.
+The integrity protection is done separately for each of the eight 32b sub-words within a 256b word.
+
+The register files can consume integrity-protected data, or add protection to unprotected data.
+Whenever possible, integrity-protected data is consumed by the register files, in particular in the following cases:
+
+* Data coming from the data memory (DMEM) through the load-store unit.
```

Comment:
```suggestion
* Data coming from the data memory (DMEM) through the load-store unit to a WDR or GPR.
```

As you said in the other comment (that github won't let me reply to) a TODO or similar here noting not all of these are implemented would be a good idea. I can think of at least one more:

* Data coming from the call stack to a GPR.

Reply:
I made these modifications and added a TODO to the ones that remain unimplemented after #6374 goes in.

Reply:
Have just realised my bullet point is wrong! The issue being there's no instruction that directly transfers from call stack to/from a GPR, as it'll all go via the ALU (e.g. `addi x2, x1, 0` would be used to pop the call stack into `x2`).

We could special case things, e.g. so `addi xn, xy, 0` does preserve integrity.

I was also thinking if we had integrity on the PC with the same code we'd carry that directly to/from the call stack but of course how PC integrity is dealt with remains an open question.